source/extensions/filters/http/aws_lambda/sts_credentials_provider.cc

```diff
@@ -84,12 +84,13 @@ void StsCredentialsProviderImpl::setWebToken(std::string_view web_token) {
 }
 
 void StsCredentialsProviderImpl::onResult(
-    std::shared_ptr<const StsCredentials> result, std::string role_arn,
-    std::list<std::string> &chained_requests) {
-    auto updated = credentials_cache_.emplace(role_arn, result);
-    if (updated.second){
-      credentials_cache_[role_arn] = result;
-    }
+  std::shared_ptr<const StsCredentials> result, std::string role_arn,
+  std::list<std::string> &chained_requests) {
+  auto inserted = credentials_cache_.emplace(role_arn, result);
+  // emplace does not override so check if inserted and if not then overrid
+  if (!inserted.second){
```

Comment:
can we add a unit test for this logic?

Reply:
Yep! Added to the full flow test. It calls this function and makes sure that the resulting cached creds are updated.